spec/lib/chip/service_spec.rb

```diff
@@ -241,4 +241,193 @@
       end
     end
   end
+
+  describe '#post_patient_check_in' do
+    let(:service_obj) { subject.build(options) }
+    let(:tenant_name) { 'mobile_app' }
+    let(:tenant_id) { Settings.chip[tenant_name].tenant_id }
+    let(:username) { 'test_username' }
+    let(:password) { 'test_password' }
+
+    context 'successful check-in' do
+      let(:response_message) do
+        'success with appointmentIen: test-appt-ien, patientDfn: test-patient-ien, stationNo: test-station-no'
+      end
+      let(:response_body) do
+        {
+          'data' => {
+            'attributes' => {
+              'message' => response_message
+            },
+            'type' => 'AuthenticatedCheckinResponse'
+          }
+        }
+      end
+
+      before do
+        expect(StatsD).to receive(:increment).once.with('api.chip.get_token.total')
+        expect(StatsD).to receive(:increment).once.with('api.chip.post_patient_check_in.total')
+      end
+
+      it 'returns 200 with success message' do
+        VCR.use_cassette('chip/authenticated_check_in/post_check_in_200') do
+          VCR.use_cassette('chip/token/token_200') do
+            response = service_obj.post_patient_check_in('test-appt-ien', 'test-patient-ien', 'test-station-no')
+            expect(response.status).to be 200
+            expect(JSON.parse(response.body)).to eq(response_body)
+          end
+        end
+      end
+    end
+
+    context 'invalid appointment response' do
+      let(:response_message) do
+        'Check-in unsuccessful with appointmentIen: appt-ien, patientDfn: patient-ien, stationNo: station-no'
+      end
+      let(:invalid_appt_response) do
+        {
+          'data' => {
+            'attributes' => {
+              'errors' => ['e-check-in-not-allowed'],
+              'message' => response_message
+            },
+            'type' => 'AuthenticatedCheckinResponse'
+          }
+        }
+      end
+
+      before do
+        expect(StatsD).to receive(:increment).once.with('api.chip.get_token.total')
+        expect(StatsD).to receive(:increment).once.with('api.chip.post_patient_check_in.total')
+      end
+
+      it 'returns 200 with error message' do
+        VCR.use_cassette('chip/authenticated_check_in/post_check_in_invalid_appointment_200') do
+          VCR.use_cassette('chip/token/token_200') do
+            response = service_obj.post_patient_check_in('test-appt-ien', 'test-patient-ien', 'test-station-no')
+            expect(response.status).to be 200
+            expect(JSON.parse(response.body)).to eq(invalid_appt_response)
+          end
+        end
+      end
+    end
+
+    context 'invalid demographics response' do
```

Comment:
I'm not sure if these specs are adding any value since this and the next spec responses are exactly like previous one and they are coming straight from the mock.

Reply:
Addressed in 086b37